fontforge/splineutil2.c

```diff
@@ -3654,12 +3654,12 @@ void SplinePointListSet(SplinePointList *tobase, SplinePointList *frombase) {
 	    tsp->noprevcp = fsp->noprevcp;
 	    tsp->nonextcp = fsp->nonextcp;
 	    if ( first==NULL ) first = tsp;
-	    if ( tsp->next==NULL )
+	    if ( tsp->next==NULL || fsp->next==NULL )
 	break;
 	}
 	firstsp = NULL;
 	for ( tspline = tobase->first->next, fspline=frombase->first->next;
-		tspline!=NULL && tspline!=firstsp;
+		tspline!=NULL && tspline!=firstsp && frombase->spiro_cnt==0;
```

Comment:
I don't understand the point of the new condition. Does it change during the iterations?